### PR TITLE
Implemented Room Db for revoked Certificates

### PR DESCRIPTION
**Pull request by gallmann-ubique**

### CODE REVIEW

File: sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/data/room/RevokedCertificateEntity.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package ch.admin.bag.covidcertificate.sdk.android.data.room
```
Comment (benz-ubique) [inline sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/data/room/RevokedCertificateEntity.kt:1]: headers missing
/*
 * Copyright (c) 2021 Ubique Innovation AG <https://www.ubique.ch>
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * SPDX-License-Identifier: MPL-2.0
 */

File: sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/data/room/RevokedCertificatesDao.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package ch.admin.bag.covidcertificate.sdk.android.data.room
```
Comment (benz-ubique) [inline sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/data/room/RevokedCertificatesDao.kt:1]: headers missing
/*
 * Copyright (c) 2021 Ubique Innovation AG <https://www.ubique.ch>
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * SPDX-License-Identifier: MPL-2.0
 */

File: sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/data/room/RevokedCertificatesDb.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package ch.admin.bag.covidcertificate.sdk.android.data.room
```
Comment (benz-ubique) [inline sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/data/room/RevokedCertificatesDb.kt:1]: headers

File: sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/repository/TrustListRepository.kt [hunk 4]
```diff
@@ -124,37 +123,32 @@ internal class TrustListRepository(
...
-			var validDuration: Long? = null
 
 			// Get the revocation list as long as the request is successful
-			var count = 0
 			var revocationListResponse = revocationService.getRevokedCertificates(getCacheControlParameter(forceRefresh), since)
 			while (revocationListResponse.isSuccessful) {
-				// Add all revoked certificates from the response to the list and save the validity duration of this response
-				allRevokedCertificates.addAll(revocationListResponse.body()?.revokedCerts.orEmpty())
-				validDuration = revocationListResponse.body()?.validDuration
+
```
Comment (benz-ubique) [inline sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/repository/TrustListRepository.kt:131]: @M-Wong  can you review this change here please

Comment (M-Wong) [reply sdk/src/main/java/ch/admin/bag/covidcertificate/sdk/android/repository/TrustListRepository.kt:131]: Looks good to me

**Review by M-Wong — APPROVED** [review]

**Review by benz-ubique — APPROVED** [review]

**Pull request merged**